beacon-chain/rpc/apimiddleware/endpoint_factory.go

```diff
@@ -0,0 +1,223 @@
+			GetResponse:           &blockHeadersResponseJson{},
+			Err:                   &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/headers/{block_id}":
+		endpoint = gateway.Endpoint{
+			GetResponse: &blockHeaderResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/blocks":
+		endpoint = gateway.Endpoint{
+			PostRequest: &beaconBlockContainerJson{},
+			Err:         &gateway.DefaultErrorJson{},
+			Hooks: gateway.HookCollection{
+				OnPostDeserializeRequestBodyIntoContainer: []gateway.Hook{prepareGraffiti},
+			},
+		}
+	case "/eth/v1/beacon/blocks/{block_id}":
+		endpoint = gateway.Endpoint{
+			GetResponse: &blockResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+			Hooks: gateway.HookCollection{
+				CustomHandlers: []gateway.CustomHandler{handleGetBlockSsz},
+			},
+		}
+	case "/eth/v1/beacon/blocks/{block_id}/root":
+		endpoint = gateway.Endpoint{
+			GetResponse: &blockRootResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/blocks/{block_id}/attestations":
+		endpoint = gateway.Endpoint{
+			GetResponse: &blockAttestationsResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/pool/attestations":
+		endpoint = gateway.Endpoint{
+			GetRequestQueryParams: []gateway.QueryParam{{Name: "slot"}, {Name: "committee_index"}},
+			GetResponse:           &attestationsPoolResponseJson{},
+			PostRequest:           &submitAttestationRequestJson{},
+			Err:                   &submitAttestationsErrorJson{},
+			Hooks: gateway.HookCollection{
+				OnPostStart: []gateway.Hook{wrapAttestationsArray},
+			},
+		}
+	case "/eth/v1/beacon/pool/attester_slashings":
+		endpoint = gateway.Endpoint{
+			PostRequest: &attesterSlashingJson{},
+			GetResponse: &attesterSlashingsPoolResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/pool/proposer_slashings":
+		endpoint = gateway.Endpoint{
+			PostRequest: &proposerSlashingJson{},
+			GetResponse: &proposerSlashingsPoolResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/beacon/pool/voluntary_exits":
+		endpoint = gateway.Endpoint{
+			PostRequest: &signedVoluntaryExitJson{},
+			GetResponse: &voluntaryExitsPoolResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/node/identity":
+		endpoint = gateway.Endpoint{
+			GetResponse: &identityResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/node/peers":
+		endpoint = gateway.Endpoint{
+			GetResponse: &peersResponseJson{},
+			Err:         &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/node/peers/{peer_id}":
+		endpoint = gateway.Endpoint{
+			GetRequestUrlLiterals: []string{"peer_id"},
+			GetResponse:           &peerResponseJson{},
+			Err:                   &gateway.DefaultErrorJson{},
+		}
+	case "/eth/v1/node/peer_count":
+		endpoint = gateway.Endpoint{
+			Path:        "/eth/v1/node/peer_count",
```

Comment:
Curious, why does this Endpoint have a path parameter but many other routes do not?

Reply:
this is a mistake